docs/DeveloperGuide.md

```diff
@@ -1,29 +1,194 @@
+specifying an index for which book to mark, or gives a negative number, an appropriate error message will be shown and the command will be rendered
+invalid. 
+
+The sequence diagram below shows how user input is processed to carry out the add book command.
+![AddCommand.png](UML_diagrams/AddCommand.png)
+
+#### Implementation and Rationale
+The `ParserMain` class incorporates exception handling to detect invalid or unrecognized commands. This allows the program to continue running
+while prompting the user for valid input
+
+By abstracting out the parsing functionality of BookBuddy into a separate `ParserMain` class, the complexity of parsing 
+user input is removed from the main code. It is instead replaced by a simple interface for the user to work with, adhering to the abstraction
+concept of object-oriented programming.
+
+### FileStorage Class Component
+The `FileStorage` class is crucial for managing file operations in BookBuddy, such as reading from and writing to files, 
+thereby ensuring data is saved for future sessions.
+
+#### Overview
+The `FileStorage` class interacts with the `BookList` and `BookListModifier` classes to load and save book data. It ensures that 
+the data directory and file exist upon initialization and provides methods for reading from and writing to the data file. 
+The class diagram below illustrates the relationship between `FileStorage` and other classes.
+
+![FileStorage.png](UML_diagrams/FileStorage.png)
+
+#### Detailed Workflow
+Upon instantiation, the `FileStorage` class checks for the existence of a predefined directory (./data) and file (books.txt),
+creating them if they do not exist. Existing data, if any, is read from the file and BookBuddy's BookList is initialized
+with this data. For reading data, it scans each line of the file, converting each line of text into instances of `Book` with
+the appropriate details. For saving data, it iterates through the BookList, converting each book into a string format suitable 
+for file storage and writes this to the file.
+
+#### Implementation and Rationale
+The creation of the `FileStorage` class ensures that all features related to reading and writing data is taken away from other
+parts of the application. This separation makes the `FileStorage` class solely responsible for these features, adhering to
+the Single Responsibility Principle.
 
 ## Product scope
 ### Target user profile
 
-{Describe the target user profile}
+Users that want an all-in-one app to track the books read, progress for each book.
+Progress for each book can be recorded according to the number of pages read.
+Users will be able to sort books according to genre.
+Users can sort books according to Read or Unread.
+Users will also be able to search for books via keywords in book titles
 
 ### Value proposition
 
 {Describe the value proposition: what problem does it solve?}
```

Comment:
Perhaps you could add a value proposition so that developers can understand at a glance what your program can offer.